src/polymer/elements/iron-form-v1-to-v2.ts

```diff
@@ -0,0 +1,135 @@
+/**
+ * @license
+ * Copyright (c) 2018 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import * as dom5 from 'dom5';
+import {Document, ParsedHtmlDocument, Replacement, Severity, Warning} from 'polymer-analyzer';
+
+import {HtmlRule} from '../../html/rule';
+import {getIndentationInside} from '../../html/util';
+import {registry} from '../../registry';
+import {stripIndentation} from '../../util';
+
+const p = dom5.predicates;
+const isIronFormV1 = p.AND(
+    (node: dom5.Node) =>
+        !node.parentNode || !p.hasTagName('iron-form')(node.parentNode),
+    p.hasTagName('form'),
+    p.hasAttrValue('is', 'iron-form'));
+
+const propertiesToDelete = [
+  'is',
+  'disable-native-validation-ui',
+  'disable-native-validation-ui$',
+  'request',
+  'request$',
+];
+const propertiesToMove = [
+  'id',
+  'id$',
+  'headers',
+  'headers$',
+  'with-credentials',
+  'with-credentials$'
+];
+const propertiesToRename: {[key: string]: string;} = {
+  'content-type': 'enctype',
+  'content-type$': 'enctype$',
+};
+
+class IronFormV1ToV2 extends HtmlRule {
+  code = 'iron-form-v1-to-v2';
+  description = stripIndentation(`
+      Warns when \`iron-form\` is used as type extension. 
+
+      This:
+
+          <form is="iron-form"
+                method="get"
+                action="/my-end-point"
+                on-iron-form-error="handleError">
+            <input type="text">
+            <input type="submit">
+          </form>
+
+      Should instead be written as:
+
+          <iron-form on-iron-form-error="handleError">
+            <form method="get" action="/my-end-point">
+              <input type="text">
+              <input type="submit">
+            </form>
+          </iron-form>
+  `);
+
+  async checkDocument(parsedDocument: ParsedHtmlDocument, _document: Document) {
+    const warnings: Warning[] = [];
+    const forms = dom5.queryAll(
+        parsedDocument.ast, isIronFormV1, [], dom5.childNodesIncludeTemplate);
```

Comment:
Changes in dom5.queryAll:

* The third parameter is dropped entirely
* It returns an iterable rather than an array
* Should be a bit faster!